File: languages/go/test/e2e_test/data_manipulation.go
package sdk_test

import (
	"os"
	"strings"

	sdk "github.com/bitwarden/sdk-go"
)

func RequiredEnv(key string) string {
	val := os.Getenv(key)
	if val == "" {
		panic("Missing required environment variable: " + key)
	}

	return val
}

func WithRunId(s string) string {
	return s + "-" + RequiredEnv("RUN_ID")
}

func FilterProjectsToThisRun(projects *sdk.ProjectsResponse) []sdk.ProjectResponse {
	var filteredProjects []sdk.ProjectResponse
	runId := RequiredEnv("RUN_ID")
	for _, project := range projects.Data {
		if strings.HasSuffix(project.Name, runId) {
			filteredProjects = append(filteredProjects, project)
		}
	}
	return filteredProjects
}

func FilterSecretsToThisRun(secrets *sdk.SecretIdentifiersResponse) []sdk.SecretIdentifierResponse {
	var filteredSecrets []sdk.SecretIdentifierResponse
	runId := RequiredEnv("RUN_ID")
	for _, secret := range secrets.Data {
		if strings.HasSuffix(secret.Key, runId) {
			filteredSecrets = append(filteredSecrets, secret)
		}
	}
	return filteredSecrets
}

File: languages/js/e2e-test/src/data-manipulation.ts
import { ProjectsResponse, SecretIdentifiersResponse } from "@bitwarden/sdk-client";

export function env(key: string) {
  const value = process.env[key];
  if (!value) {
    throw new Error(`Missing environment variable: ${key}`);
  }
  return value;
}

export function withRunId(s: string) {
  return `${s}-${env("RUN_ID")}`;
}

export function filterProjectsToThisRun(projects: ProjectsResponse) {
  if (!projects?.data) {
    throw new Error("No projects found");
  }
  return projects.data.filter((project) => project.name.endsWith(env("RUN_ID")));
}

export function filterSecretsToThisRun(secrets: SecretIdentifiersResponse) {
  if (!secrets?.data) {
    throw new Error("No secrets found");
  }
  return secrets.data.filter((secret) => secret.key.endsWith(env("RUN_ID")));
}
